core/src/test/java/org/elasticsearch/bootstrap/BootstrapForTesting.java

```diff
@@ -83,6 +83,13 @@
         if (systemPropertyAsBoolean("tests.security.manager", true)) {
             try {
                 Security.setCodebaseProperties();
+                // if its an insecure plugin, its not easy to simulate here, since we don't have a real plugin install.
+                // we just do our best so unit testing can work. integration tests for such plugins are essential.
+                String artifact = System.getProperty("tests.artifact");
+                String insecurePluginProp = Security.INSECURE_PLUGINS.get(artifact);
+                if (insecurePluginProp != null) {
+                    System.setProperty(insecurePluginProp, "file:/-");
```

Comment:
It means everything! So the comment is really true, proper integration tests are essential. 

Our unit testing framework for plugins (loading via classpath) is just not good enough. And when that runs the plugin's code and third party dependencies are spread everywhere so we cant formulate a URL pattern for it.

I don't think we should spend time making security stuff complicated or elaborate for tests that will never be realistic anyway. Better to spend the effort fixing the problematic third party libs?